### PR TITLE
[ci skip] added explanation to link_directory and link_tree directives

### PR DESCRIPTION
**Pull request by udaykadaboina**
- Added explanation to `link_directory` and `link_tree` directives.
- Fixed typos
- Added the directive details to end_user_asset_generation.md file.

### CODE REVIEW

File: guides/end_user_asset_generation.md [hunk 1]
```diff
@@ -151,34 +151,61 @@ Now in your `application.js` will correctly load the `foo.min.js` before `foo-ui
...
 ### require_self
-TODO: add description for `require_self`
+
+`require_self` tells Sprockets to insert the body of the current
+source file before any subsequent `require` directives.
 
 ### require_tree
-TODO: add description for `require_tree`
+
+`require_tree` *path* works like `require_directory`, but operates
```
Comment (rafaelfranca) [inline guides/end_user_asset_generation.md]: We should move this to after `require_directory`

File: README.md [hunk 2]
```diff
@@ -404,6 +404,16 @@ automatically be defined for you.
 }
 ```
 
+#### The `link_directory` Directive
+
+`link_directory` *path* links all the files inside the directory specified by the *path*
+
+#### The `link_tree` Directive
+
+`link_tree` *path* works like `link_directory`, but operates
```
Comment (connorshea) [inline README.md]: I think it'd make more sense to have `link_tree PATH` so it's highlighted.

Comment (connorshea) [inline README.md]: Alternatively, `link_tree path/to/asset`

File: guides/end_user_asset_generation.md (hunk 1 above)
Comment (udaykadaboina) [inline guides/end_user_asset_generation.md]: Thanks @rafaelfranca will make the changes and will submit PR.

**Comment by udaykadaboina** [comment]
Hey @rafaelfranca let me know how this looks now. :)

**Comment by schneems** [comment]
Thanks!

**Pull request merged**